### PR TITLE
Bug 2089254: Rotate token secret if its older than 30 minutes

### PR DESCRIPTION
**Pull request by alexander-demichev**
The token secret might expire, this PR makes sure that the secret is rotated every 30 minutes.

### CODE REVIEW

File: pkg/controllers/kubeconfig/suite_test.go [hunk 1]
```diff
@@ -23,6 +24,7 @@ var (
 	cfg     *rest.Config
 	cl      client.Client
 	ctx     = context.Background()
+	timeout = 10 * time.Minute
```
Comment (JoelSpeed) [inline pkg/controllers/kubeconfig/suite_test.go:27]: 10 minute timeout?

Comment (alexander-demichev) [reply pkg/controllers/kubeconfig/suite_test.go]: probably it's too much, I replaced it with 1 minute

Comment (JoelSpeed) [reply pkg/controllers/kubeconfig/suite_test.go]: Normally I would expect to use something like 10 seconds, 1 minute for the controller to react would indicate an issue to me and surely means the watches aren't working right

Comment (alexander-demichev) [reply pkg/controllers/kubeconfig/suite_test.go]: 👍done

**Pull request merged**